### PR TITLE
[jvm-packages] default objective to binary:logistic for XGBoostClassifier

### PR DESCRIPTION
**Pull request by wbo4958**
Currently, the default objective for both XGBoostClassifier and
XGBoostRegressor is "reg:squarederror". 

So this PR is to default objective to binary:logistic for XGBoostClassifier.

### CODE REVIEW

**Comment by wbo4958** [comment]
@CodingCat Could you help to review this PR?

**Comment by trivialfis** [comment]
How does it work when objective is supposed to be `multi:softmax`?

**Comment by wbo4958** [comment]
> How does it work when objective is supposed to be `multi:softmax`?

Hi trivialfis. the previous default value is to default classifier objective to be regressor objective. which seems not to make sense. 

This PR defaults classifier objective to be classifier related objective, we can default to multi:softmax, that' ok.

For your question, it's not working. suppose it will throw exception.

**Comment by CodingCat** [comment]
Why we have to make softmax as default?

**Comment by wbo4958** [comment]
> Why we have to make softmax as default?
No, we don't

**Comment by trivialfis** [comment]
On Python side we actually choose between these 2 objs depending on `num_class`.  It caused some headaches (output of logistic doesn't have `prob` and `max`) but still it works.  Let's figure out a way that can be more general.
